src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static updateAlives() {
+    Animal.alive = Animal.alive.filter((el) => el.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  unhide() {
+    this.hidden = false;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+      Animal.updateAlives();
+    }
```

Comment:
The bite handling is a Herbivore behavior and therefore, it would be considered a best practice to implement it inside the `Herbivore` class.
As an optional task, could you think of a way to move this logic into the Herbivore class?